mash_client/cli/config.py

```diff
@@ -119,7 +119,59 @@ def setup_config(
 ):
     """
-    Creates a client config file based on user input to prompts.
+    Create a configuration file for the mash command line tool
+
+    .SH OPTIONS
+    .TP
+    \\fB--color\\fP Colorize the output
+    By default the output is not colorized. Using \\fI--color\\fP as an option
+    will enable output colorization.
+    .TP
+    \\fB--config-dir\\fP PATH_TO_CONFIGURATION_DIR
+    By default the configuration file is written to the users home directory
+    in \\fIconfig/mash_client/\\fP. There is one configuration file per user
+    profile. This allows a given user to configure access to multiple mash
+    server instances and/or access to multiple mash users on the mash server.
+    Whether or not multiple user accounts for the same user are supported on
+    the mash server is a function of the mash server configuration.
+    If the configuration file is in a non default location
+    it must be specified with each invocation of a \\fImash command\\fP
+    .TP
+    \\fB--email\\fP MASH_USER_EMAIL
+    Depending on the server configuration this setting may not be needed.
+    The mash server supports single sign on via OIDC. In this case your mash
+    server user account will be setup with the e-mail from the OIDC server and
+    all authentication is token based.
+    .TP
+    \\fB--host\\fP MASH_SERVER_PROTOCOL_AND_URI
+    The mash client communicates with the mash server using the http protocol.
+    Depending on the server setup encrypted communication, i.e. https, is also
+    supported. The \\fIhost\\fP and \\fIport\\fP option work together in that
+    it is possible to specify a server using configured for https as using the
+    http protocol on port 443 (if the server runs TLS on the default port) or
+    specifying the host as using https, for example 'https://192.168.1.95'.
+    In this configuration the \\fIport\\P setting is not necessary.
+    .TP
+    \\fB--log-level\\fP JOB_DEFAULT_LOG_LEVEL
+    Set the log level to use for submitted jobs. Log level values follow the
+    Python log setting, https://docs.python.org/3/library/logging.html#levels
+    .TP
+    \\fB--port\\fP SERVER_PORT_FOR_API
+    The \\fIport\\P and \\fIhost\\fP work together to form the communication
+    path to the mash server. The protocol used is always http and depending
+    on the server setup and the protocol specification in the \\fIhost\\fP
+    setup it may not be necessary to specify the \\fIport\\P. If the server
+    uses default ports, 80, or 443, simply specify the protocl as part of the
+    \\fIhost\\fP setting.
+    .TP
+    \\fB--profile\\fP THE_PROFILE_NAME_FOR_THIS_CONFIGURATION
+    Each profile is stored in a separate YAML file in the specified
+    configuration directory. The specified profile name becomes the filename
+    for the stored configuration. Case sensitivity of the profile name is
+    determined by the underlying filesystem.
+    .TP
+    \\fB--verify\\fP
```

Comment:
We can link to the requests documentation but it can also be confusing. This is a multi type option. It can be boolean whether or not to verify SSL cert of server. Or it can be a path to a chain cert. If it's a path to cert then it's by default "set to True" and verification happens using the supplied chain cert. It can also be a a directory to certs but this is a use case I have not considered handling nor tested.

Reply:
https://2.python-requests.org/en/v2.8.1/user/advanced/#ssl-cert-verification